tests/test_templates.py

```diff
@@ -16,12 +14,16 @@
 from test_project import redirect_stdout, redirect_stderr
 
 
-class BaseTemplateTest(object):
+class TestBaseTemplate(object):
     project_class = signac.Project
-    env = None
+    envm = None
+
+    for name, env in flow.environment.ComputeEnvironment.registry.items():
```

Comment:
This loop should be inside your test function `test_get_TestEnvironment` and use `subtests.test` for every environment.